src/ElasticaService.php

```diff
@@ -471,4 +464,52 @@ protected function exception($exception)
         );
         $this->logger->error($message);
     }
+
+    /**
+     * Check if response has any errors
+     *
+     * @param Response|null $response
+     * @throws Exception
+     */
+    protected function logResponse(Response $response = null)
+    {
+        // Ignore empty or non-error responses
+        if (!$response || $response->isOk()) {
+            return;
+        }
+
+        // Get error message
+        $data = $response->getData();
+        $errorMessage = $data['message']
+            ?? $response->getErrorMessage()
+                ?: sprintf("HTTP %d error", $response->getStatus());
+        $message = "Elastica server error: $errorMessage";
+
+        // If no logger specified expose error normally
+        if (!$this->logger) {
+            throw new Exception($message, $response->getStatus());
+        }
+
+        // Log message
+        $this->logger->error($message);
+    }
+
+    /**
+     * Run elastic search query
+     *
+     * @param callable $callback A callback that generates an Elastica Response object
+     * @return Response The response
+     * @throws Exception
+     */
+    protected function runQuery(callable $callback)
+    {
+        $response = null;
+        try {
+            $response = call_user_func($callback);
+            $this->logResponse($response);
+        } catch (Exception $ex) {
+            $this->exception($ex);
+        }
+        return $response;
```

Comment:
Minor point - newlines between var declarations, `return`s and logic.

Reply:
I suggest to setup a linter and run a github action; I normally setup phpcs with a standard psr ruleset, verified via github actions run on commit.

Reply:
Fixed